frontend/src/metabase/query_builder/components/filters/FilterPopover/FilterPopover.unit.spec.tsx

```diff
@@ -154,4 +158,43 @@ describe("FilterPopover", () => {
       });
     });
   });
+  describe("filter rendering", () => {
+    it("should not render filter picker when filter type is 'Is empty'", async () => {
+      const filter = new Filter(IS_EMPTY_FILTER, null, QUERY);
+      renderWithProviders(
+        <FilterPopover
+          query={QUERY}
+          filter={filter}
+          onChangeFilter={dummyFunction}
+        />,
+      );
+
+      expect(screen.getByTestId("select-button-content")).toHaveTextContent(
+        "Is empty",
+      );
+
+      expect(
+        screen.getByTestId("default-picker-container").childNodes.length,
+      ).toBe(0);
+    });
+
+    it("should render a space for a filter if filter is not 'empty' or 'not empty'", async () => {
+      const filter = new Filter(STRING_CONTAINS_FILTER, null, QUERY);
+      renderWithProviders(
+        <FilterPopover
+          query={QUERY}
+          filter={filter}
+          onChangeFilter={dummyFunction}
+        />,
+      );
+
+      expect(screen.getByTestId("select-button-content")).toHaveTextContent(
+        "Contains",
+      );
+
+      expect(
+        screen.getByTestId("default-picker-container").childNodes.length,
+      ).toBeGreaterThan(0);
```

Comment:
I don't understand the name of this test or this assertion.
Can you please describe what is this checking?

Reply:
We're checking if the number of children within the `default-picker-container` `div` is 0 - i.e. there is no picker component (i.e. no text box, no dropdown, etc)

Reply:
Removed this since the test is a little more generic and just checks the separator. More tests can be made in the future for each of the different pickers